tests/test_access_token.py

```diff
@@ -86,14 +89,15 @@ def test_access_token(client):
     nonce = get_nonce(client, cons_wallet.address)
     _hash = add_ethereum_prefix_and_hash_msg(f'{ddo.did}{nonce}')
     payload['signature'] = Web3Helper.sign_hash(_hash, cons_wallet)
-    payload['delegateAddress'] = 'thisIsNotAValidAddress'  # noqa
+    payload.pop('delegatePublicKey')
     request_url = at_endpoint + '?' + '&'.join(
         [f'{k}={v}' for k, v in payload.items()]
     )
     response = client.get(request_url)
+    # missing delegatePublicKey
     assert response.status_code == 400, f'{response.data}'
 
-    payload['delegateAddress'] = some_wallet.address
+    payload['delegatePublicKey'] = get_private_key(some_wallet).public_key
```

Comment:
please make sure public_key is a string not an object, or at least that it resolves to str when dumped to json.

Reply:
'&'.join([f'{k}={v}' for k, v in payload.items()]) will perform an automatic string conversion on this object, and is indeed what I recieve through the payload. Do you want me to add an extra conversion step?

Reply:
perfect.